src/recorder/jamrecorder.h

```diff
@@ -178,6 +178,7 @@ class CJamRecorder : public QObject
 
 signals:
     void RecordingSessionStarted ( QString sessionDir );
+    void RecordingFailed ( QString error );
```

Comment:
This would become `CGenErr error`.

Reply:
As discussed, I'm not changing that for now as I'm having troubles with exactly that declaration (maybe due to lack of such a constructor?). The alternative would be passing by reference, which I'm not sure is a safe thing to do from catch blocks and to pass on via signals.